### PR TITLE
Improve doc test coverage (java)

### PR DESCRIPTION
**Pull request by mlopatkin**
Adding tests. The only more or less substantial change is the fix for the `javadoc` configuration.

### CODE REVIEW

**Comment by mlopatkin** [comment]
@bot-gradle test PRF, BD

**Comment by mlopatkin** [comment]
@bot-gradle test PRF

File: subprojects/docs/src/snippets/java/apt/kotlin/build.gradle.kts [hunk 1]
```diff
@@ -9,9 +9,9 @@ repositories {
 // tag::annotation-processing[]
 dependencies {
     // The dagger compiler and its transitive dependencies will only be found on annotation processing classpath
-    annotationProcessor("com.google.dagger:dagger-compiler:2.8")
+    annotationProcessor("com.google.dagger:dagger-compiler:2.44")
 
     // And we still need the Dagger library on the compile classpath itself
-    implementation("com.google.dagger:dagger:2.8")
+    implementation("com.google.dagger:dagger:2.44")
```
Comment (6hundreds) [inline subprojects/docs/src/snippets/java/apt/kotlin/build.gradle.kts:15]: BTW why do we have to decrease the version?

**Review by 6hundreds — APPROVED** [review]
LGTM, with a tiny question

File: subprojects/docs/src/snippets/java/apt/kotlin/build.gradle.kts (hunk 1 above)
Comment (mlopatkin) [reply subprojects/docs/src/snippets/java/apt/kotlin/build.gradle.kts:15]: It is an increase, `8 < 44` :). Dagger 2.8 is not compatible with Java 11, so I had to bump the version.

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

File: subprojects/docs/src/snippets/java/apt/kotlin/build.gradle.kts (hunk 1 above)
Comment (6hundreds) [reply subprojects/docs/src/snippets/java/apt/kotlin/build.gradle.kts:15]: Ah I see, probably too late for review :)

**Review by bamboo — COMMENTED** [review]
👍

**Pull request merged**